contracts/src/v0.8/dev/VRFV2Wrapper.sol

```diff
@@ -0,0 +1,260 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.4;
+
+import "../vendor/SafeMathChainlink.sol";
+
+import "../ConfirmedOwner.sol";
+import "../interfaces/TypeAndVersionInterface.sol";
+import "../VRFConsumerBaseV2.sol";
+import "../interfaces/LinkTokenInterface.sol";
+import "../interfaces/AggregatorV3Interface.sol";
+import "../interfaces/VRFCoordinatorV2Interface.sol";
+import "../vendor/SafeMathChainlink.sol";
+import "./VRFV2WrapperInterface.sol";
+
+/**
+ * @notice A wrapper for VRFCoordinatorV2 that provides an interface better suited to one-off
+ * @notice requests for randomness.
+ */
+contract VRFV2Wrapper is ConfirmedOwner, TypeAndVersionInterface, VRFConsumerBaseV2, VRFV2WrapperInterface {
+  using SafeMathChainlink for uint256;
+
+  LinkTokenInterface public immutable LINK;
+  AggregatorV3Interface public immutable LINK_ETH_FEED;
+  ExtendedVRFCoordinatorV2Interface public immutable COORDINATOR;
+  uint64 public immutable SUBSCRIPTION_ID;
+
+  // lastRequestId is the request ID of the most recent VRF V2 request made by this wrapper. This
+  // should only be relied on within the same transaction the request was made.
+  uint256 public override lastRequestId;
+
+  // Configuration fetched from VRFCoordinatorV2
+
+  // s_configured tracks whether this contract has been configured. If not configured, randomness
+  // requests cannot be made.
+  bool public s_configured;
+
+  // s_disabled disables the contract when true. When disabled, new VRF requests cannot be made
+  // but existing ones can still be fulfilled.
+  bool public s_disabled;
+
+  // s_fallbackWeiPerUnitLink is the backup LINK exchange rate used when the LINK/NATIVE feed is
+  // stale.
+  int256 private s_fallbackWeiPerUnitLink;
```

Comment:
Can't pull this from the coordinator directly? Similar question for the other configuration fields below.

Reply:
Right yeah, if coordinator config changes we need to make a call to update the config here too. Could get the config each time but adds more gas cost, I'm open to either. In the end though, this should only affect profitability so I'd rather take the gas savings for users